src/Logary/Logger.fs

```diff
@@ -1,94 +1,82 @@
 ﻿namespace Logary
 
-/// Main interface used to log LogLines and Metrics.
+open Logary.DataModel
+
 type Logger =
   inherit Named
 
   /// Write a Verbose log line
-  abstract LogVerbose : (unit -> LogLine) -> unit
+  abstract LogVerbose : (unit -> Message) -> unit
 
   /// Write a Debug log line
-  abstract LogDebug   : (unit -> LogLine) -> unit
+  abstract LogDebug   : (unit -> Message) -> unit
 
   /// Write a log line to the logger.
-  abstract Log        : LogLine -> unit
+  abstract Log        : Message -> unit
 
   /// Write a measure to the logger.
-  abstract Measure    : Measure -> unit
+  abstract Measure    : Message -> unit
 
   /// Gets the currently set log level, aka. the granularity with which things
   /// are being logged
   abstract Level      : LogLevel
 
-/// API for writing logs and measures.
-/// For gauges that run continuously based on a timer, have a look at
-/// the Registry module. To get a logger, have a look at the Logger module.
-///
-/// If you are using I recommend doing:
-/// `Install-Package Logary.CSharp` instead of dealing with the
-/// interop problems that you will get from using this module directly.
 [<CompilationRepresentation(CompilationRepresentationFlags.ModuleSuffix)>]
 module Logger =
   open System
 
   open Logary
-  open Logary.LogLine
-
-  let private ensurePath (loggerName : string) (line : LogLine) =
-    if String.IsNullOrEmpty line.path then
-      LogLine.setPath loggerName line
-    else line
+  open Logary.DataModel
 
   /////////////////////
   // Logging methods //
   /////////////////////
 
+  let private setContext (logger : Logger) (msg : Message) =
+    Message.Context.serviceSet logger.Name msg
```

Comment:
Now that I've thought about it more, it's completely wrong. It should set the `name` instead of the `service`, and not overwite it if it's already set.